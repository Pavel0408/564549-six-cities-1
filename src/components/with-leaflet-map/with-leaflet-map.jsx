import leaflet from "leaflet";
import {PureComponent} from "react";
import PropTypes from "prop-types";

export class WithLeafletMap extends PureComponent {


  render() {
    return this.props.render({createMap: this.createMap, addOffersPins: this.addOffersPins});
  }

  createMap(props) {

    const {city, zoom, settings} = props;

    const map = leaflet.map(settings.id, settings.options);
    map.setView(city, zoom);

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    return map;
  }

  addOffersPins({offers, mapOffers, icon}) {
    offers.map((offer) => {
      const offerCords = offer.coordinates;
      return leaflet
        .marker(offerCords, {icon})
        .addTo(mapOffers);
    });
  }
}

WithLeafletMap.propTypes = {
  render: PropTypes.func.isRequired
};


