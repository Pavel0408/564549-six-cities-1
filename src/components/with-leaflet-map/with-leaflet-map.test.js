import React from 'react';
import renderer from 'react-test-renderer';

import {WithLeafletMap} from "./with-leaflet-map";

const render = () => {
  return <div id="map" style={{height: 823}}></div>;
};

describe(`testing WithLeafletMap render`, () => {
  it(`WithLeafletMap is render correctly`, () => {
    const tree = renderer.create(<WithLeafletMap
      render={render}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
