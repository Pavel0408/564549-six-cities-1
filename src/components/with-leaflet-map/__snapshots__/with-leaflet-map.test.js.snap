// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing WithLeafletMap render WithLeafletMap is render correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": 823,
    }
  }
/>
`;
